lib/checkmemoryleak.cpp

```diff
@@ -996,7 +996,7 @@ void CheckMemoryLeakNoVar::checkForUnreleasedInputArgument(const Scope *scope)
         const Token* tok2 = tok->next()->astParent();
         while (tok2 && tok2->isCast())
             tok2 = tok2->astParent();
-        if (tok2 && tok2->isAssignmentOp())
+        if (tok2 && (tok2->isAssignmentOp() || tok2->str() == "return"))
```

Comment:
I prefer `Token::Match(tok2, "%assign%|return")`

Reply:
The null pointer check is not required with `Token::Match`, right?